### ISSUE TITLE
Model.create with callback leads to unhandled rejection

### ISSUE THREAD

**Issue by atomfried** [Automattic/mongoose#4269]
Hi there,

in my understanding Model.create invoked with a callback like so

```
thingModel.create(newThing, cb);
```

should not create a promise, however it seems like it does because the above leads to messages like

```
Unhandled rejection ValidationError: [...]
```

which go away when adding a catch like so

```
thingModel.create(newThing, cb).catch(cb2);
```

which is confusing, because according to the doc examples _cb_ receives an error argument, too.

I'm new to JS, so please correct me if I'm wrong.

Thanks,

M

**Comment by vkarpov15** [comment]
Yeah when you set mongoose to use bluebird you will always get the unhandled rejection warning because mongoose will return a promise even when you pass a callback. This is something we want to improve in a future release

**Comment by atomfried** [comment]
Okay, so I'll use the promise for now. Thanks for your answer.

**Comment by niftylettuce** [comment]
Possibly related @vkarpov15, #4364?

**Comment by vkarpov15** [comment]
Duplicate of #3670, fixed in 5.0 branch :+1:
